extensions/optaplanner/deployment/src/main/java/io/quarkus/optaplanner/deployment/OptaPlannerProcessor.java

```diff
@@ -0,0 +1,250 @@
+        }
+        String constraintProviderClassName = classInfos.iterator().next().name().toString();
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        try {
+            // Don't use recorderContext.classProxy(constraintProviderClassName)
+            // because ReflectiveClassBuildItem cannot cope with a class proxy
+            return (Class<? extends ConstraintProvider>) classLoader.loadClass(constraintProviderClassName);
+        } catch (ClassNotFoundException e) {
+            throw new IllegalStateException("The constraintProviderClass (" + constraintProviderClassName
+                    + ") cannot be created during deployment.", e);
+        }
+    }
+
+    private void applyTerminationProperties(SolverConfig solverConfig) {
+        TerminationConfig terminationConfig = solverConfig.getTerminationConfig();
+        if (terminationConfig == null) {
+            terminationConfig = new TerminationConfig();
+            solverConfig.setTerminationConfig(terminationConfig);
+        }
+        optaPlannerQuarkusConfig.solver.termination.spentLimit.ifPresent(terminationConfig::setSpentLimit);
+        optaPlannerQuarkusConfig.solver.termination.unimprovedSpentLimit.ifPresent(terminationConfig::setUnimprovedSpentLimit);
+        optaPlannerQuarkusConfig.solver.termination.bestScoreLimit.ifPresent(terminationConfig::setBestScoreLimit);
+    }
+
+    private String convertAnnotationInstancesToString(Collection<AnnotationInstance> annotationInstances) {
+        return "[" + annotationInstances.stream().map(instance -> instance.target().toString())
+                .collect(Collectors.joining(", ")) + "]";
+    }
+
+    private String convertClassInfosToString(Collection<ClassInfo> classInfos) {
+        return "[" + classInfos.stream().map(instance -> instance.name().toString())
+                .collect(Collectors.joining(", ")) + "]";
+    }
+
+    @BuildStep
+    public RuntimeInitializedClassBuildItem nativeImageDroolsTricks() {
+        return new RuntimeInitializedClassBuildItem(ClassFieldAccessorFactory.class.getName());
+    }
+
+    // TODO health check
+
+    @BuildStep
+    void registerOptaPlannerJacksonModule(BuildProducer<AdditionalBeanBuildItem> additionalBeans,
+            Capabilities capabilities) {
+        if (!capabilities.isCapabilityPresent(Capabilities.JACKSON)) {
+            return;
+        }
+        try {
+            Class.forName("org.optaplanner.persistence.jackson.api.OptaPlannerJacksonModule", false,
+                    Thread.currentThread().getContextClassLoader());
+        } catch (Exception e) {
+            // Fail fast during build to avoid a certain runtime failure
+            throw new IllegalStateException(
+                    "When using both Jackson and OptaPlanner,"
+                            + " add a dependency on org.optaplanner:optaplanner-persistence-jackson too.",
```

Comment:
Yea - the idea is that it would stick out like a soar thumb, fail-fast basically. While if you forgot to add quarkus-optaplanner-jackson, it's hard to notice that score instances aren't being jacksoned correctly.
Anyway, not worth the hassle - I am creating quarkus-optaplanner-jackson to align with the quarkus ecosystem spirit :)

Reply:
Done